x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -148,8 +166,10 @@ export const useInitSourcerer = (
             id === scopeDataViewId
               ? selectedPatterns.length === 0 && missingPatterns.length === 0
               : timelineDataViewId === id
-              ? timelineMissingPatterns.length === 0 && timelineSelectedPatterns.length === 0
+              ? timelineMissingPatterns.length === 0 &&
+                timelineSelectedDataView?.patternList.length === 0
               : false,
+          skipScopeUpdate: timelineSelectedPatterns.length > 0 && timelineDataViewId === id,
```

Comment:
If the data view needs to be init (get fields/pattern list) but the sourcerer scope is already set to the data view and has a saved pattern list (timeline), then do not update the selected data view